lib/model/unreads.dart

```diff
@@ -286,14 +306,23 @@ class Unreads extends ChangeNotifier {
                     .add(messageId);
               }
             }
-            newlyUnreadInStreams.forEach((incomingStreamId, incomingTopics) {
-              incomingTopics.forEach((incomingTopic, incomingMessageIds) {
+            for (
+              final MapEntry(key: incomingStreamId, value: incomingTopics)
+              in newlyUnreadInStreams.entries
+            ) {
+              for (
+                final MapEntry(key: incomingTopic, value: incomingMessageIds)
+                in incomingTopics.entries
+              ) {
                 _addAllInStreamTopic(incomingMessageIds..sort(), incomingStreamId, incomingTopic);
```

Comment:
I wish I had a good idea for making these more concise, but I don't. :-/ Short of, like, an extension `entryPairs` on `Map` to return 2-tuples instead of `MapEntry` — which is probably what `Map.entries` would have done in the first place if tuples/records had existed in Dart from early on.

Can at least make them more compact vertically, though:
```suggestion
            for (final MapEntry(key: incomingStreamId, value: incomingTopics)
                 in newlyUnreadInStreams.entries) {
              for (final MapEntry(key: incomingTopic, value: incomingMessageIds)
                   in incomingTopics.entries) {
                _addAllInStreamTopic(incomingMessageIds..sort(), incomingStreamId, incomingTopic);
```